staging/src/k8s.io/api/flowcontrol/v1alpha1/types.go

```diff
@@ -0,0 +1,414 @@
+	// +optional
+	DistinguisherMethod *FlowDistinguisherMethod `json:"distinguisherMethod,omitempty" protobuf:"bytes,3,opt,name=distinguisherMethod"`
+	// `rules` describes which requests will match this flow schema. This FlowSchema matches a request if and only if
+	// at least one member of rules matches the request.
+	// +optional
+	Rules []PolicyRulesWithSubjects `json:"rules" protobuf:"bytes,4,rep,name=rules"`
+}
+
+// FlowDistinguisherMethodType is the type of flow distinguisher method
+type FlowDistinguisherMethodType string
+
+// These are valid flow-distinguisher methods.
+const (
+	// FlowDistinguisherMethodByUserType specifies that the flow distinguisher is the username in the request.
+	// This type is used to provide some insulation between users.
+	FlowDistinguisherMethodByUserType FlowDistinguisherMethodType = "ByUser"
+
+	// FlowDistinguisherMethodByNamespaceType specifies that the flow distinguisher is the namespace of the
+	// object that the request acts upon. If the object is not namespaced, or if the request is a non-resource
+	// request, then the distinguisher will be the empty string. An example usage of this type is to provide
+	// some insulation between tenants in a situation where there are multiple tenants and each namespace
+	// is dedicated to a tenant.
+	FlowDistinguisherMethodByNamespaceType FlowDistinguisherMethodType = "ByNamespace"
+)
+
+// FlowDistinguisherMethod specifies the method of a flow distinguisher.
+type FlowDistinguisherMethod struct {
+	// `type` is the type of flow distinguisher method
+	// The supported types are "ByUser" and "ByNamespace".
+	// Required.
+	Type FlowDistinguisherMethodType `json:"type" protobuf:"bytes,1,opt,name=type"`
+}
+
+// PriorityLevelConfigurationReference contains information that points to the "request-priority" being used.
+type PriorityLevelConfigurationReference struct {
+	// `name` is the name of resource being referenced
+	// Required.
+	Name string `json:"name" protobuf:"bytes,1,opt,name=name"`
+}
+
+// PolicyRulesWithSubjects prescribes a test that applies to a request to an apiserver. The test considers the subject
+// making the request, the verb being requested, and the resource to be acted upon. This PolicyRulesWithSubjects matches
+// a request if and only if both (a) at least one member of subjects matches the request and (b) at least one member
+// of resourceRules or nonResourceRules matches the request.
+type PolicyRulesWithSubjects struct {
+	// `subjects` is the list of normal user, serviceaccount, or group that this rule cares about.
+	// There must be at least one subject for this field, or it will be rejected by the api validation.
+	// Required.
+	Subjects []Subject `json:"subjects" protobuf:"bytes,1,rep,name=subjects"`
+	// `resourceRules` is a slice of ResourcePolicyRules that identify matching requests according to their verb and the
+	// resource they request to act upon.
```

Comment:
it is less precise, but more understandable.

Reply:
"... that identify matching requests according to their verb and the resource they request to act upon" is close to a garden-path sentence for me.

Reply:
If others think it is clear enough, I am OK with "target" rather than the longer way of saying it.  But if anybody thinks "target resource" or "target object" might refer to a ServiceAccount in our Subjects list then we are in trouble.

As noted elsewhere, we should say "object" rather than "resource", this should say "object" rather than "resource" because we do not want to rule out discriminating more finely.  In fact, the KEP exhibits some rules that discriminate on the namespace of the target object.

Reply:
> s/verb and the resource they request to act upon/verb and target resource/

updated the comment

Reply:
I see no update here.